extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -497,22 +497,18 @@ export async function makeHttpRequest<B>(
 	} else if (response.status < 200 || response.status > 299) {
 		let errorMessage: string[] = [];
 		errorMessage.push(response.status.toString());
-		if (response.data && response.data.error) {
-			errorMessage.push(`${response.data.error.code} : ${response.data.error.message}`);
+		const data = response.data;
+		if (isErrorResponseBodyWithError(data)) {
+			errorMessage.push(`${data.error.code} : ${data.error.message}`);
 		}
 		const error = new Error(errorMessage.join(EOL));
 		if (!ignoreErrors) {
 			throw error;
 		}
 		result.errors.push(error);
-	}
-
-	if (response) {
-		result.response = {
-			headers: response.headers,
-			data: response.data.body,
```

Comment:
This is where it was broken - I had it returning the response wrapped in another HttpClientResponse - but that wasn't what was actually being returned by the calls to the http client.